token-cli: more better errors in command_transfer

2501babe opened the pull request:
check if the user needs to provide a memo
dont let the user create ata accounts belonging to other token-owned accounts

closes #3711

2501babe commented on token/cli/src/main.rs:
```diff
@@ -4290,7 +4338,7 @@ mod tests {
 
         let token = create_token(&config, &payer).await;
         let source = create_associated_account(&config, &payer, token).await;
-        let recipient = token.to_string();
+        let recipient = Keypair::new().pubkey().to_string();
```
i changed this because the errors i added to `command_transfer` break it. however i think this was done just as a convenience, and there isnt a real usecase for automatically creating an associated account owned by a mint

joncinque reviewed (commented):
Just little nits, otherwise, looks great! Could you add a test for the new error behavior?

2501babe commented on token/cli/src/main.rs:
```diff
@@ -230,6 +231,26 @@ fn is_multisig_minimum_signers(string: String) -> Result<(), String> {
+    match is_pubkey(string.as_ref()) {
+        Ok(()) => {
+            let program_id = string.as_ref().parse::<Pubkey>().unwrap();
+            if VALID_TOKEN_PROGRAM_IDS
+                .iter()
+                .any(|valid_id| program_id == *valid_id)
```
i knew there was a function for this but couldnt remember it :sob:

2501babe commented:
i added test cases for all the new possible errors, consolidated the system flag tests and fixed them to use both programs, and went ahead and converted all tests from hardcoded lists of program ids to using the new static

joncinque reviewed (approved):
Looks great! The test coverage is perfect
